src/middleware/FileRestrictionsMiddleware.ts

```diff
@@ -0,0 +1,94 @@
+import { SessionKey } from 'ch-node-session-handler/lib/session/keys/SessionKey';
+import { SignInInfoKeys } from 'ch-node-session-handler/lib/session/keys/SignInInfoKeys';
+import { UserProfileKeys } from 'ch-node-session-handler/lib/session/keys/UserProfileKeys';
+import { ISignInInfo, IUserProfile } from 'ch-node-session-handler/lib/session/model/SessionInterfaces';
+import { NextFunction, Request, Response } from 'express';
+import { FORBIDDEN } from 'http-status-codes';
+import { provide } from 'inversify-binding-decorators';
+import { BaseMiddleware } from 'inversify-express-utils';
+import { loggerInstance } from './Logger';
+
+import { Appeal } from 'app/models/Appeal';
+import { AppealsPermissionKeys } from 'app/models/AppealPermissionKeys';
+import { APPLICATION_DATA_KEY } from 'app/models/ApplicationData';
+import { Attachment } from 'app/models/Attachment';
+
+const customErrorTemplate = 'error-custom';
+
+@provide(FileRestrictionsMiddleware)
+export class FileRestrictionsMiddleware extends BaseMiddleware {
+
+    public handler(req: Request, res: Response, next: NextFunction): void {
+
+        const session = req.session;
+
+        const adminPermissionFlag = session.chain(_ => _.getValue<ISignInInfo>(SessionKey.SignInInfo))
+            .chainNullable(signInInfo => signInInfo[SignInInfoKeys.AdminPermissions])
+            .extract();
+
+        const userProfile: IUserProfile = session.chain(_ => _.getValue<ISignInInfo>(SessionKey.SignInInfo))
+            .chainNullable(signInInfo => signInInfo[SignInInfoKeys.UserProfile])
+            .ifNothing(() => loggerInstance().error(`${FileRestrictionsMiddleware.name} - User profile was expected in session but none found`))
+            .unsafeCoerce();
+
+        const permissions = userProfile[UserProfileKeys.Permissions];
+
+        if (adminPermissionFlag === '1') {
+            if (permissions &&
+                permissions[AppealsPermissionKeys.download] &&
+                permissions[AppealsPermissionKeys.view]) {
+                return next();
+            }
+            return this.renderForbiddenError(res);
+
+        } else {
+
+            const appeal: Appeal = session.chain(_ => _.getExtraData())
+                .chainNullable(extraData => extraData[APPLICATION_DATA_KEY])
+                .chainNullable(applicationData => applicationData.appeal)
+                .ifNothing(() => loggerInstance().error(`${FileRestrictionsMiddleware.name} - Appeal was expected in session but none found`))
+                .unsafeCoerce();
+
+            const fileId = req.params.fileId;
+            const attachement = this.getAttachmentFrom(appeal, fileId);
+
+            if (!attachement) {
+                return this.renderForbiddenError(res);
+            }
+
+            // User must be creating a new appeal.
+            if (!appeal.createdBy) {
```

Comment:
there should always be a 'createdBy'.. we could check this earlier on? If there is no 'createdBy', should this return an error?

Reply:
where do we set the createdBy on the frontend?

Reply:
We could set it on the frontend at the start of the journey, I don't see why not tbf

Reply:
createdBy is set by the API, but you're right by saying that if this appeal data hasn't come from the API, there will not be a createdBy value. So if there is no createdBy value, we allow the user to download the document as long as the attachment exists?

Reply:
Yep, I think there is no risk in doing that. The user can't download files that are not from that appeal object and can't access other appeals that aren't the users if the user tries to manipulate urls. I've been drawing out scenarios where the user could exploit that but I still haven't found one